src/odb.c

```diff
@@ -75,6 +75,10 @@ static int odb_read_hardcoded(git_rawobj *raw, const git_oid *id)
 	raw->type = type;
 	raw->len = 0;
 	raw->data = git__calloc(1, sizeof(uint8_t));
+
+	if (!raw->data)
+		return -1;
```

Comment:
Why don't you use `GITERR_ALLOC_CHECK` here?

Reply:
At the moment, `odb_read_hardcoded` simply returns `-1` to indicate that the file was not found in the hardcoded object list.  I'll change this to actually distinguish between found/notfound and error separately.